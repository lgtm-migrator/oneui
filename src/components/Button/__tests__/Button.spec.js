import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../Button';

describe('<Button> that renders a button', () => {
    const buttonTitle = 'Click me';

    it('should render default button correctly', () => {
        const { container } = render(<Button>{buttonTitle}</Button>);
        expect(container).toMatchSnapshot();
    });
    it('should add classes when props are changed', () => {
        const { container } = render(
            <Button context="primary" size="large" isBlock>
                {buttonTitle}
            </Button>
        );
        expect(container).toMatchSnapshot();
    });
    it('should call click callback correctly', () => {
        const onClickMock = jest.fn();
        const { getByText } = render(<Button onClick={onClickMock}>{buttonTitle}</Button>);
        fireEvent.click(getByText(buttonTitle));
        expect(onClickMock).toHaveBeenCalled();
    });
    it('should add string html attributes correctly', () => {
        const { container } = render(<Button data-test="something">{buttonTitle}</Button>);
        expect(container.querySelector('button')).toHaveAttribute('data-test', 'something');
    });
    it('should add functional html attributes correctly', () => {
        const onMouseOverMock = jest.fn();
        const { container } = render(
            // eslint-disable-next-line jsx-a11y/mouse-events-have-key-events
            <Button onMouseOver={onMouseOverMock}>{buttonTitle}</Button>
        );
        fireEvent.mouseOver(container.querySelector('button'));
        expect(onMouseOverMock).toHaveBeenCalled();
    });
    it('should render an ancor element if href is defined', () => {
        const { container } = render(<Button href="/">{buttonTitle}</Button>);
        expect(container).toMatchSnapshot();
        expect(container.querySelector('a')).toBeVisible();
    });
});
