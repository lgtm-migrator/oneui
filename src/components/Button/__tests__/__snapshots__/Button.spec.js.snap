// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> that renders a button should add classes when props are changed 1`] = `
<div>
  <button
    class="Button Button--context_primary Button--size_large Button--isBlock"
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`<Button> that renders a button should render an ancor element if href is defined 1`] = `
<div>
  <a
    class="Button Button--context_neutral"
    href="/"
  >
    Click me
  </a>
</div>
`;

exports[`<Button> that renders a button should render default button correctly 1`] = `
<div>
  <button
    class="Button Button--context_neutral"
    type="button"
  >
    Click me
  </button>
</div>
`;
